packages/blade/src/components/Box/__tests__/useMemoizedStyles.web.test.tsx

```diff
@@ -15,3 +18,27 @@ describe('getDependencyProp', () => {
     ).toMatchInlineSnapshot(`"{\\"paddingLeft\\":\\"12px\\",\\"display\\":\\"block\\"}"`);
   });
 });
+
+describe('useMemoizedStyles', () => {
+  const MockComponentToTestHook = ({
+    returnValueCallback,
+  }: {
+    returnValueCallback: (val: CSSObject) => void;
+  }): null => {
+    const memoVal = useMemoizedStyles({
+      padding: 'spacing.10',
+      margin: ['spacing.1', 'spacing.2'],
+      theme: paymentLightTheme,
+    });
+    returnValueCallback(memoVal);
+    return null;
+  };
+
+  it('should return correct CSS styles', () => {
```

Comment:
so to test this shouldn't we mock the screen width and then again assert that the values should be same until we hit the breakpoint and this function shouldn't be called till the breakpoint is changed? that will give us the true result? you can refer `useBreakpoint.web.test`

Reply:
So we don't mock screen size change in `useBreakpoint`. We only mock `window.matchMedia` which is different from actually changing the screen size.

In `useBreakpoint`, we then mock addEventListeners of `window.matchMedia`. 

I did try adding that code here but the event listener itself isn't get triggered because we're not using `window.matchMedia` directly anywhere in Box's implementation. 

<img width="716" alt="image" src="https://user-images.githubusercontent.com/30949385/225218536-554d0a73-de80-4c99-8ae8-b72436907232.png">